tools/CMakeLists.txt

```diff
@@ -160,3 +160,21 @@ if(RET EQUAL 0)
 else()
     message(FATAL_ERROR "File generation failed")
 endif()
+
+
+# The files need to be generated with cmake as they are needed for the linker
+# command to be added 
+execute_process(
+    COMMAND ${Python3_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/tfm_parse_manifest_list.py
+    -m ${COMBINED_LIST}
+    -f ${CMAKE_CURRENT_SOURCE_DIR}/tfm_debug_symbol_list.yaml
+    -l ${TFM_ISOLATION_LEVEL}
+    -b ${CONFIG_TFM_SPM_BACKEND}
+    -o ${CMAKE_BINARY_DIR}/generated
+    ${PARSE_MANIFEST_QUIET_FLAG}
+    RESULT_VARIABLE RET
+)
+
+if(NOT RET EQUAL 0)
+    message(FATAL_ERROR "Symbol File generation failed, debug info may be missing")
```

Comment:
I think that the error message does not match with the error level. The error message as I read it implies that the building will continue without symbols. But since you classified it as FATAL_ERROR this will cause the building to stop. I would say better classify it as a warning and keep the error message.

Reply:
If this fails, then the file is missing, to read out the functions in secure_fw/CMakeLists.txt, causing then the build to fail. 
but probably there is a way to prevent this from failing.

Reply:
I agree that we can't simply classify as warning as that would just trigger a more obscure error later in the build.

FATAL_ERROR is OK. Although it would be more future-proof to have this configurable and enabled by default in case this machinery causes an unexpected problem and we need a quick workaround to disable it.

A configuration option can be introduced by adding a line to
trusted-firmware-m/config/config_default.cmake

Reply:
I changed it to a warning, and made the  part for adding the linker option fail-safe if the file doesn't exist